### PR TITLE
Ruff: enable most of bugbear

### PR DESCRIPTION
**Pull request by jnewb1**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -197,7 +197,7 @@ build-backend = "poetry.core.masonry.api"
 
 # https://beta.ruff.rs/docs/configuration/#using-pyprojecttoml
 [tool.ruff]
-select = ["E", "F", "W", "PIE", "C4", "ISC", "RUF100"]
+select = ["E", "F", "W", "PIE", "C4", "ISC", "RUF100", "B006"]
```
Comment (adeebshihadeh) [inline pyproject.toml:200]: think we want all the B's

Comment (jnewb1) [reply pyproject.toml]: Added all the B's except a few I marked below

File: system/hardware/tici/hardware.py [hunk 2]
```diff
@@ -83,6 +83,17 @@ def affine_irq(val, action):
   for i in irqs:
     sudo_write(str(val), f"/proc/irq/{i}/smp_affinity_list")
 
+@lru_cache
```
Comment (adeebshihadeh) [inline system/hardware/tici/hardware.py:86]: what's the reason on this one? seems worse

File: selfdrive/athena/athenad.py [hunk 3]
```diff
@@ -213,6 +213,16 @@ def retry_upload(tid: int, end_event: threading.Event, increase_count: bool = Tr
         break
 
 
+def cb(sm, item, tid, sz: int, cur: int) -> None:
```
Comment (adeebshihadeh) [inline selfdrive/athena/athenad.py:216]: same here

File: system/hardware/tici/hardware.py (hunk 2 above)
Comment (jnewb1) [reply system/hardware/tici/hardware.py:86]: # lru_cache and cache can cause memory leaks when used in classes

Comment (jnewb1) [reply system/hardware/tici/hardware.py:86]: https://beta.ruff.rs/docs/rules/cached-instance-method/

File: selfdrive/athena/athenad.py (hunk 3 above)
Comment (jnewb1) [reply selfdrive/athena/athenad.py:216]: https://beta.ruff.rs/docs/rules/function-uses-loop-variable/

File: system/hardware/tici/hardware.py (hunk 2 above)
Comment (adeebshihadeh) [reply system/hardware/tici/hardware.py:86]: oh, this happens to be fine here, but seems like a good rule to have

File: selfdrive/athena/athenad.py (hunk 3 above)
Comment (jnewb1) [reply selfdrive/athena/athenad.py:216]: I can also just move the function up before the loop rather than to the global scope

**Review by adeebshihadeh — APPROVED** [review]

**Pull request merged**